### ISSUE TITLE
Does using model.save() and then using model.load() resume training exactly from the point where it was left?

### ISSUE THREAD

**Issue by myidispg** [DLR-RM/stable-baselines3#29]
I am not worried about the previous metric like rewards etc. Does model.load() also load the previous optimizer parameters too?

**Comment by araffin** [comment]
Hello,
it is not properly documented yet but the answer is in the roadmap: https://github.com/DLR-RM/stable-baselines3/issues/1

So yes, but you need to save the replay buffer too (cf zoo) if you are using one.

**Comment by myidispg** [comment]
I was using the PPO algorithm and just took a look at the zip file it saves. It has `policy.pth` and `policy.optimizer.pth` in it which must be the Neural Network and the Optimizer state dictionary. Am I right? Also, what does the `tensors.pth` file contain?

**Comment by Miffyli** [comment]
Yup, that sounds about right. I am personally going over the review of code regarding these (sorry for slowness, arrafin ^^'). The tensors seem to contain misc objects, but I am not sure why these are called tensors (one point in my review currently). This object should be an empty list for PPO, but I think this is a typo and should not exist if it is empty.

**Comment by araffin** [comment]
>The tensors seem to contain misc objects, but I am not sure why these are called tensors 

so it is mostly meant for additional pytorch tensors (hence the name) because they are not part of the policy and require special care when saving and loading (we had issues because of cpu/gpu saving/loading).

>This object should be an empty list for PPO, but I think this is a typo and should not exist if it is empty 

@Artemis-Skade did that part, but yes, PPO has not special tensors to save.

**Comment by myidispg** [comment]
Thanks for the prompt replies. I trained a PPO model on Google Colab yesterday but I had to halt the process so I saved it and when I tried resuming it today, I noticed there was a very sharp decline in the reward obtained.

Here is the code snippet for resuming the training:
```
model = PPO('MlpPolicy', env, verbose=0, learning_rate=3e-5)
model.load('/content/drive/My Drive/DQN_Models/stable_baselines/ppo_11v11_easy')
model.set_env(env)
model.learn(total_timesteps=int(1e7), callback = callback)
```

Here is the 'rewards obtained' graph at every episode from yesterday: 
![image](https://user-images.githubusercontent.com/36075176/82745545-4a4cd200-9da3-11ea-9fa1-2972ee911de1.png)

Here is the graph when I loaded the model and observed a decline in the 'rewards obtained':
![download (1)](https://user-images.githubusercontent.com/36075176/82745650-7e74c280-9da4-11ea-9c13-36160dc95875.png)

Is this expected?
